lib/singleton.rb

```diff
@@ -156,14 +147,16 @@ def append_features(mod)
       unless mod.instance_of?(Class)
         raise TypeError, "Inclusion of the OO-Singleton module in module #{mod}"
       end
+
       super
     end
 
     def included(klass)
       super
+
       klass.private_class_method :new, :allocate
       klass.extend SingletonClassMethods
-      Singleton.__init__(klass)
+      klass.instance_eval { @singleton__mutex__ = Thread::Mutex.new }
```

Comment:
`instance_variable_set` may be preferable to `instance_eval` here, since only a single instance variable is set.